assets/js/modules/analytics/components/common/ExistingGTMPropertyNotice.js

```diff
@@ -27,15 +27,22 @@ import { sprintf, __ } from '@wordpress/i18n';
 import Data from 'googlesitekit-data';
 import { MODULES_ANALYTICS } from '../../datastore/constants';
 import { MODULES_TAGMANAGER } from '../../../tagmanager/datastore/constants';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
 const { useSelect } = Data;
 
 export default function ExistingGTMPropertyNotice() {
 	const propertyID = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS ).getPropertyID()
 	);
 
-	const gtmAnalyticsPropertyID = useSelect( ( select ) =>
-		select( MODULES_TAGMANAGER ).getSingleAnalyticsPropertyID()
+	const isTagManagerAvaliable = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleAvailable( 'tagmanager' )
+	);
+
+	const gtmAnalyticsPropertyID = useSelect(
+		( select ) =>
+			isTagManagerAvaliable &&
```

Comment:
@hussain-t Shall we just return `false` when TagManager is not available, which will prevent the call to `getSingleAnalyticsPropertyID` from going through? Right now, even if TagManager is not available, looks like `select( MODULES_TAGMANAGER ).getSingleAnalyticsPropertyID()` will still be fired.

Reply:
@asvinb, `getSingleAnalyticsPropertyID` call will be fired only if the `isTagManagerAvailable` evaluates to `true`. Otherwise, the call will not be fired hence the `useSelect` will return `false`. It's the same as explicitly returning `false` like this:

```js
if ( ! isTagManagerAvailable ) {
	return false;
}

return select( MODULES_TAGMANAGER ).getSingleAnalyticsPropertyID();
```

Reply:
Ah you're right! Thanks @hussain-t 👍